### DISCUSSION TITLE
How do I set the heartbeat period of socket.io-client:1.0.0

### DISCUSSION THREAD

**Discussion by huangjinbing** [socketio/socket.io#4856] [Q&A] [answered]
![49d6b3d8fd50834aa072c147fe5a873](https://github.com/socketio/socket.io/assets/13633806/7beaffd9-2224-4528-9665-64390c5599c1)
I can't find a place to set pingInterval，Please help. Thank you

**Comment by darrachequesne** [comment, marked as answer]
Hi! The `pingInterval` option is managed by the server, not by the client.

Reference: https://socket.io/docs/v4/server-options/#pinginterval

**Reply by huangjinbing** [reply]
Thank you for your answer, but I hope some clients can be modified to reduce device power consumption

**Reply by huangjinbing** [reply]
If it is controlled by the server, all clients are affected

**Reply by darrachequesne** [reply]
There is currently no way to use a specific value per client. This could be updated in the future though, if there's enough interest.
